R/coh_data_wrangling.R

```diff
@@ -352,9 +363,14 @@ get_time_to_event <- function(data, outcome_date_col,
     colnames(data),
     must.include = outcome_date_col
   )
-  # check strings, converted to date later
-  checkmate::assert_string(start_cohort)
-  checkmate::assert_string(end_cohort)
+  # check date types
+  checkmate::assert_date(start_cohort)
+  checkmate::assert_date(end_cohort)
+
+  # check if date columns are date type
+  checkmate::assert_date(
+    data[[outcome_date_col]]
+  )
```

Comment:
Not sure whether you should allow missing values in this column, would be good to check with Santiago. If not, add a check here using `any.missing = FALSE`.

Reply:
Missing values are allowed in all the date-type columns. In the cohort study, the population is split into those with and without outcome, based on whether they have or not a date reported.